docs/Developers/Writing_bear_actions.rst

```diff
@@ -0,0 +1,174 @@
+      is the file in the state when result was generated.
+   -  ``file_diff_dict``: A dictionary whose key is the filename and value
+      is a diff of file from the state in the original_file_dict to
+      the current state.
+   -  ``applied_actions``: List of actions names that have already been applied
+      for the current result.
+
+   .. note::
+
+       ``is_applicable`` doesn’t have to be a static method. In this case you
+       also need to prepend ``self`` to the parameters in the signature.
+
+2. ``apply`` method:
+   It has the logic for when the action is applied.
+
+   .. code:: python
+
+           def apply(self,
+                     result,
+                     original_file_dict,
+                     file_diff_dict,
+                     **kwargs):
+               """
+               Apply (S)ome Action
+               """
+
+   apply method must return ``file_diff_dict``.
+
+   Writing a docstring for ``apply`` method is very important. The contents of
+   the docstring will get displayed to the user when asked to choose an action
+   to apply. The letter in parentheses can be used by the user as a way of
+   selecting this action.
+
+Apart from this you can also add other helper methods and even an ``init``
+method to the action class.
+
+Passing bear action to the result
+---------------------------------
+
+After writing bear action you have to let coala know about this.
+When a bear yields a result you can pass a list action instances for that
+bear by using an optional argument ``actions``.
+
+Suppose ``SomeAction`` is an action specific to ``SomeBear``
+
+.. code:: python
+
+    from path.to.SomeAction import SomeAction
+
+    class SomeBear:
+
+        def run():
+
+            yield Result('origin', 'message', actions=[SomeAction()])
+
+Example of EditCommitMessageAction
+----------------------------------
+
+We will look at an example of ``EditCommitMessageAction`` for
+``GitCommitBear``. Whenever GitCommitBear suggests some changes in commit
+message we can provide an action to the user, which on applying
+will open up an editor where user can edit the commit message.
+
+.. code:: python
+
+    import subprocess
+    from coalib.results.result_actions.ResultAction import ResultAction
+
+    class EditCommitMessageAction(ResultAction):
+
+        SUCCESS_MESSAGE = 'Commit message edited successfully.'
+
+        def apply(self, result, original_file_dict, file_diff_dict):
+            """
+            Edit (C)ommit Message [Note: This may rewrite your commit history]
+            """
+            subprocess.check_call(['git', 'commit', '-o', '--amend'])
+            return file_diff_dict
+
+Note that ``is_applicable`` method is not implemented, that is because
+``ResultAction`` already implements this and it always returns True which
+is what we what want in this case.
+
+``apply`` method spawns a simple process which runs a git command which will
+open up an editor to edit the commit message.
```

Comment:
This is not related to this issue.

Reply:
I don't remember editing this files but I will take a look at my code immediately.